src/alert/ztf.rs

```diff
@@ -254,6 +214,39 @@ where
     deserialize_isdiffpos_option(deserializer).map(|x| x.unwrap())
 }
 
+fn deserialize_fid<'de, D>(deserializer: D) -> Result<String, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    // the fid is a mapper: 1 = g, 2 = r, 3 = i
+    let fid: i32 = serde::Deserialize::deserialize(deserializer)?;
+    match fid {
+        1 => Ok("g".to_string()),
+        2 => Ok("r".to_string()),
+        3 => Ok("i".to_string()),
+        _ => Err(serde::de::Error::custom(format!("Unknown fid: {}", fid))),
+    }
+}
+
+fn deserialize_ssnamenr<'de, D>(deserializer: D) -> Result<Option<String>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    // if the value is null, "null", "", return None
+    let value: serde_json::Value = serde::Deserialize::deserialize(deserializer)?;
+    match value {
+        serde_json::Value::Null => Ok(None),
+        serde_json::Value::String(s) => {
+            if s.is_empty() || s.eq_ignore_ascii_case("null") {
+                Ok(None)
+            } else {
+                Ok(Some(s))
+            }
+        }
+        _ => Ok(None),
+    }
```

Comment:
I haven't tested it, but I offer this alternative:

```rust
    let value: Option<String> = serde::Deserialize::deserialize(deserializer)?;
    Ok(value.filter(|s| !s.is_empty() && !s.eq_ignore_ascii_case("null")))
```

Reply:
added!